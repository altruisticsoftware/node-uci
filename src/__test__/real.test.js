/* eslint-disable no-console */

import { Engine } from '../'
import Promise from 'bluebird'

const enginePath = 'engines/stockfish-9-64'

describe('real engine', () => {
  describe('promise/async', () => {
    it('promise/async usage', async () => {
      const engine = new Engine(enginePath)
      await engine.init()
      expect(engine.options).toMatchSnapshot()
      await engine.setoption('MultiPV', '4')
      expect(engine.options).toMatchSnapshot()
      const go = await engine.go({ depth: 4 })
      expect(go).toBeDefined()
      await engine.quit()
    })

    it('goinfinite usage', async () => {
      /* eslint-disable-next-line no-undef */
      jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000

      const engine = new Engine(enginePath)
      await engine.init()
      engine.goInfinite()
      await Promise.delay(5000)
      const bm = await engine.stop()
      expect(bm.bestmove).toBeDefined()
      expect(bm.info).toBeDefined()
      await engine.quit()
    })
  })

  xdescribe('chain', () => {
    it('chain usage', done => {
      const engine = new Engine(enginePath)
      engine
        .chain()
        .init()
        .setoption('MultiPV', 3)
        .position(
          'r1bqkbnr/pppp1ppp/2n5/1B2p3/4P3/5N2/PPPP1PPP/RNBQK2R b KQkq - 3 3',
        )
        .go({ depth: 15 })
        .then(result => {
          console.log('FIN')
          console.log(result)
          done()
        })
        .catch(err => {
          console.log(err.stack)
        })
    })
  })
})
