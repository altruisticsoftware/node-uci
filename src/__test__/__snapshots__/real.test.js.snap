// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`real engine promise/async promise/async usage 1`] = `
Object {
  "Clear Hash": Object {
    "type": "button",
  },
  "Contempt": Object {
    "default": 20,
    "max": 100,
    "min": -100,
    "type": "spin",
  },
  "Hash": Object {
    "default": 16,
    "max": 131072,
    "min": 1,
    "type": "spin",
  },
  "Minimum Thinking Time": Object {
    "default": 20,
    "max": 5000,
    "min": 0,
    "type": "spin",
  },
  "Move Overhead": Object {
    "default": 30,
    "max": 5000,
    "min": 0,
    "type": "spin",
  },
  "MultiPV": Object {
    "default": 1,
    "max": 500,
    "min": 1,
    "type": "spin",
  },
  "Ponder": Object {
    "default": false,
    "type": "check",
  },
  "Skill Level": Object {
    "default": 20,
    "max": 20,
    "min": 0,
    "type": "spin",
  },
  "Slow Mover": Object {
    "default": 89,
    "max": 1000,
    "min": 10,
    "type": "spin",
  },
  "Syzygy50MoveRule": Object {
    "default": true,
    "type": "check",
  },
  "SyzygyPath": Object {
    "default": "<empty>",
    "type": "string",
  },
  "SyzygyProbeDepth": Object {
    "default": 1,
    "max": 100,
    "min": 1,
    "type": "spin",
  },
  "SyzygyProbeLimit": Object {
    "default": 6,
    "max": 6,
    "min": 0,
    "type": "spin",
  },
  "Threads": Object {
    "default": 1,
    "max": 512,
    "min": 1,
    "type": "spin",
  },
  "UCI_Chess960": Object {
    "default": false,
    "type": "check",
  },
  "nodestime": Object {
    "default": 0,
    "max": 10000,
    "min": 0,
    "type": "spin",
  },
}
`;

exports[`real engine promise/async promise/async usage 2`] = `
Object {
  "Clear Hash": Object {
    "type": "button",
  },
  "Contempt": Object {
    "default": 20,
    "max": 100,
    "min": -100,
    "type": "spin",
  },
  "Hash": Object {
    "default": 16,
    "max": 131072,
    "min": 1,
    "type": "spin",
  },
  "Minimum Thinking Time": Object {
    "default": 20,
    "max": 5000,
    "min": 0,
    "type": "spin",
  },
  "Move Overhead": Object {
    "default": 30,
    "max": 5000,
    "min": 0,
    "type": "spin",
  },
  "MultiPV": Object {
    "default": 1,
    "max": 500,
    "min": 1,
    "type": "spin",
    "value": "4",
  },
  "Ponder": Object {
    "default": false,
    "type": "check",
  },
  "Skill Level": Object {
    "default": 20,
    "max": 20,
    "min": 0,
    "type": "spin",
  },
  "Slow Mover": Object {
    "default": 89,
    "max": 1000,
    "min": 10,
    "type": "spin",
  },
  "Syzygy50MoveRule": Object {
    "default": true,
    "type": "check",
  },
  "SyzygyPath": Object {
    "default": "<empty>",
    "type": "string",
  },
  "SyzygyProbeDepth": Object {
    "default": 1,
    "max": 100,
    "min": 1,
    "type": "spin",
  },
  "SyzygyProbeLimit": Object {
    "default": 6,
    "max": 6,
    "min": 0,
    "type": "spin",
  },
  "Threads": Object {
    "default": 1,
    "max": 512,
    "min": 1,
    "type": "spin",
  },
  "UCI_Chess960": Object {
    "default": false,
    "type": "check",
  },
  "nodestime": Object {
    "default": 0,
    "max": 10000,
    "min": 0,
    "type": "spin",
  },
}
`;

exports[`real engine promise/async promise/async usage 3`] = `
Object {
  "bestmove": "e2e4",
  "info": Array [
    Object {
      "depth": 1,
      "multipv": 1,
      "nodes": 74,
      "nps": 74000,
      "pv": "e2e4",
      "score": Object {
        "unit": "cp",
        "value": 112,
      },
      "seldepth": 1,
      "tbhits": 0,
      "time": 1,
    },
    Object {
      "depth": 1,
      "multipv": 2,
      "nodes": 74,
      "nps": 74000,
      "pv": "d2d4",
      "score": Object {
        "unit": "cp",
        "value": 112,
      },
      "seldepth": 1,
      "tbhits": 0,
      "time": 1,
    },
    Object {
      "depth": 1,
      "multipv": 3,
      "nodes": 74,
      "nps": 74000,
      "pv": "e2e3",
      "score": Object {
        "unit": "cp",
        "value": 100,
      },
      "seldepth": 1,
      "tbhits": 0,
      "time": 1,
    },
    Object {
      "depth": 1,
      "multipv": 4,
      "nodes": 74,
      "nps": 74000,
      "pv": "d2d3",
      "score": Object {
        "unit": "cp",
        "value": 92,
      },
      "seldepth": 1,
      "tbhits": 0,
      "time": 1,
    },
    Object {
      "depth": 2,
      "multipv": 1,
      "nodes": 171,
      "nps": 171000,
      "pv": "e2e4 b7b6",
      "score": Object {
        "unit": "cp",
        "value": 107,
      },
      "seldepth": 2,
      "tbhits": 0,
      "time": 1,
    },
    Object {
      "depth": 2,
      "multipv": 2,
      "nodes": 171,
      "nps": 171000,
      "pv": "d2d4 b7b6",
      "score": Object {
        "unit": "cp",
        "value": 97,
      },
      "seldepth": 2,
      "tbhits": 0,
      "time": 1,
    },
    Object {
      "depth": 2,
      "multipv": 3,
      "nodes": 171,
      "nps": 171000,
      "pv": "e2e3 c7c6",
      "score": Object {
        "unit": "cp",
        "value": 94,
      },
      "seldepth": 2,
      "tbhits": 0,
      "time": 1,
    },
    Object {
      "depth": 2,
      "multipv": 4,
      "nodes": 171,
      "nps": 171000,
      "pv": "d2d3 b7b6",
      "score": Object {
        "unit": "cp",
        "value": 78,
      },
      "seldepth": 2,
      "tbhits": 0,
      "time": 1,
    },
    Object {
      "depth": 3,
      "multipv": 1,
      "nodes": 391,
      "nps": 391000,
      "pv": "e2e4 d7d6 d2d4",
      "score": Object {
        "unit": "cp",
        "value": 144,
      },
      "seldepth": 3,
      "tbhits": 0,
      "time": 1,
    },
    Object {
      "depth": 3,
      "multipv": 2,
      "nodes": 391,
      "nps": 391000,
      "pv": "d2d4 e7e6 e2e4",
      "score": Object {
        "unit": "cp",
        "value": 136,
      },
      "seldepth": 3,
      "tbhits": 0,
      "time": 1,
    },
    Object {
      "depth": 3,
      "multipv": 3,
      "nodes": 391,
      "nps": 391000,
      "pv": "e2e3 e7e6 d2d4",
      "score": Object {
        "unit": "cp",
        "value": 88,
      },
      "seldepth": 3,
      "tbhits": 0,
      "time": 1,
    },
    Object {
      "depth": 3,
      "multipv": 4,
      "nodes": 391,
      "nps": 391000,
      "pv": "d2d3 e7e6 e2e4",
      "score": Object {
        "unit": "cp",
        "value": 74,
      },
      "seldepth": 3,
      "tbhits": 0,
      "time": 1,
    },
    Object {
      "depth": 4,
      "multipv": 1,
      "nodes": 965,
      "nps": 482500,
      "pv": "e2e4 e7e6 d2d4 d7d6",
      "score": Object {
        "unit": "cp",
        "value": 137,
      },
      "seldepth": 4,
      "tbhits": 0,
      "time": 2,
    },
    Object {
      "depth": 4,
      "multipv": 2,
      "nodes": 965,
      "nps": 482500,
      "pv": "d2d4 d7d5 c1g5 b7b5",
      "score": Object {
        "unit": "cp",
        "value": 89,
      },
      "seldepth": 4,
      "tbhits": 0,
      "time": 2,
    },
    Object {
      "depth": 4,
      "multipv": 3,
      "nodes": 965,
      "nps": 482500,
      "pv": "e2e3 e7e6 d2d4 d7d5",
      "score": Object {
        "unit": "cp",
        "value": 28,
      },
      "seldepth": 4,
      "tbhits": 0,
      "time": 2,
    },
    Object {
      "depth": 4,
      "multipv": 4,
      "nodes": 965,
      "nps": 482500,
      "pv": "g1f3 d7d5 d2d4 g8f6",
      "score": Object {
        "unit": "cp",
        "value": 28,
      },
      "seldepth": 4,
      "tbhits": 0,
      "time": 2,
    },
  ],
  "ponder": "e7e6",
}
`;
